### DISCUSSION TITLE
Privacy and Data Ownership: Exploring Whisper Model's Data Flow and OpenAI's Involvement

### DISCUSSION THREAD

**Discussion by mjund** [openai/whisper#1424] [Q&A]
Hey everyone,

I have a question regarding the deployment of my own Whisper model. I wanted to clarify whether the data, specifically the voice recordings from my users or the text generated by the Whisper model, has any chance of being sent to OpenAI's servers.

I value privacy and data ownership, so it's crucial for me to understand the data flow and ensure that user data remains within the confines of my deployment. If anyone has insights or information on this matter, I would greatly appreciate your input.

Thank you!

**Comment by jwnacnud** [comment]
If you have concerns, try loading it onto your local machine and then
disconnect from the internet. If it still works, then you can be sure it
isn’t “phoning home”.

On Sat, Jun 3, 2023 at 11:24 AM mazahd ***@***.***> wrote:

> I have a question regarding the deployment of my own Whisper model. I
> wanted to clarify whether the data, specifically the voice recordings from
> my users or the text generated by the Whisper model, has any chance of
> being sent to OpenAI's servers.
>
> I value privacy and data ownership, so it's crucial for me to understand
> the data flow and ensure that user data remains within the confines of my
> deployment. If anyone has insights or information on this matter, I would
> greatly appreciate your input.
>
> Thank you!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openai/whisper/discussions/1424>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGW5A7VIW57LBA7WB3Z7NLXJNXMNANCNFSM6AAAAAAYZM74HQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Jeffrey Duncan

**Reply by mjund** [reply]
maybe there is a background process like a whisper daemon or similar that handles the task of sending data if there is an available network connection, and if not, tries again later?

**Reply by phineas-pta** [reply]
the code is open source, you can read it

the model weights are available, you can inspect it

you can also monitor your connections to see which program connect owhich website

**Comment by jwnacnud** [comment]
If you are hyper concerned about making sure that your data never goes
online, run this on a machine that is not connected to the Internet. You
could also run it on a virtual machine that has no network connections.

On Sat, Jun 3, 2023 at 11:14 PM mazahd ***@***.***> wrote:

> maybe there is a background process like a whisper daemon or similar that
> handles the task of sending data if there is an available network
> connection, and if not, tries again later?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openai/whisper/discussions/1424#discussioncomment-6081364>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGW5A3F7XDAZ2MRP65DVETXJQKR5ANCNFSM6AAAAAAYZM74HQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Jeffrey Duncan

**Comment by eddywashere** [comment]
This question is referenced in other places as being an answer or at least a response to the question of data privacy. For folks who cannot review this codebase or modify the usage setup of whisper, would someone be kind enough to verify that there is no process for pushing data out to OpenAI?

example by @mjund

> maybe there is a background process like a whisper daemon or similar that handles the task of sending data if there is an available network connection, and if not, tries again later

**Comment by ryanheise** [comment]
Also see: https://github.com/openai/whisper/discussions/1462#discussioncomment-6251914
